adapters/oidc/adapter-core/src/main/java/org/keycloak/adapters/OAuthRequestAuthenticator.java

```diff
@@ -398,6 +398,9 @@ private String rewrittenRedirectUri(String originalUri) {
                 StringBuilder redirectUriBuilder = new StringBuilder(url.getProtocol());
```

Comment:
Perhaps we use something other than a `StringBuilder`? I am sure there are standard APIs available to us to build URLs?

Reply:
There is, not sure if available to adapter module (or if we want to improve this at all now).

Reply:
I think it's fine to clean this up, boy scout rule and all. Also, joining URLs manually can lead to unexpected errors, so it makes more sense to me to handle this through established methods. I already see a `URL` class being used in several places here.

Reply:
I meant that we can probably accomplish something along the likes of:

```java
Map.Entry<String, String> rule = rewriteRules.entrySet().iterator().next();
URI original = new URI(originalUri);
URI redirect = new URIBuilder(original)
        .setPath(original.getPath().replaceFirst(rule.getKey(), rule.getValue()))
        .build();

return redirect.toString();
```

Reply:
Probably or we can? :)